sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -282,6 +282,12 @@ class ResolveSessionCatalog(
       AlterTableRecoverPartitionsCommand(
         v1TableName.asTableIdentifier,
         "MSCK REPAIR TABLE")
+
+    case TruncateTableStatement(tableName, partitionSpec) =>
```

Comment:
shall we add a `truncateTable` method to `TableCatalog`? cc @rdblue

Reply:
No. This is not a catalog operation, it is a table operation. We already have 2 ways to truncate a table: if the table implements `SupportsTruncate` or `SupportsOverwrite`. A truncate command should build a write and commit without any commit messages.

We can also add a short-cut trait on `Table` to avoid the builder.